### ISSUE TITLE
 jdk17_headless still installs Zulu v16 jdk on aarch64-darwin 

### ISSUE THREAD

**Issue by shaunsingh** [NixOS/nixpkgs#145522]
### Describe the bug
On Darwin, trying to install jdk17_headless still installs Zulu version 16

### Steps To Reproduce
(on aarch64-darwin)
```
nix-shell -p jdk17_headless --command "java --version"
openjdk 16.0.1 2021-04-20
OpenJDK Runtime Environment Zulu16.30+19-CA (build 16.0.1+9)
OpenJDK 64-Bit Server VM Zulu16.30+19-CA (build 16.0.1+9, mixed mode)
```

### Expected behavior
Openjdk17 should be installed instead of the v16 Zulu jdk
Output on linux for example: 
```nix
nix shell nixpkgs/nixos-unstable#jdk17_headless
$ which java
/nix/store/y1g3vianvjp9jf25785bkjngzbrziq2n-openjdk-headless-17.0.1+12/bin/java
```

### Additional context
OpenJDK 17 should install, as it has native m1 support: https://openjdk.java.net/jeps/391

### Notify maintainers

@edwtjo 
@asbachb 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"aarch64-darwin"`
 - host os: `Darwin 21.1.0, macOS 12.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.5pre20211007_844dd90`
 - channels(shauryasingh): `"darwin"`
 - channels(root): `"nixpkgs-21.11pre327990.4789953e5c1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Artturin** [comment]
the entry in all-packages.nix is 
```
  openjdk17 =
    if stdenv.isDarwin then
      callPackage ../development/compilers/openjdk/darwin { }
    else
      callPackage ../development/compilers/openjdk/17.nix {
        openjfx = openjfx15;
        inherit (gnome2) GConf gnome_vfs;
      };
```
and openjdk/darwin is still at 16 for some reason https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/darwin/default.nix @marsam

**Comment by jiegec** [comment]
Fixed by #143836:

```shell
$ nix-shell -p jdk17_headless --command "java --version"
openjdk 17.0.3 2022-04-19 LTS
OpenJDK Runtime Environment Zulu17.34+19-CA (build 17.0.3+7-LTS)
OpenJDK 64-Bit Server VM Zulu17.34+19-CA (build 17.0.3+7-LTS, mixed mode, sharing)
```
